app/models/dialog.rb

```diff
@@ -65,6 +65,11 @@ def validate_children
       errors.add(:base, _("Dialog %{dialog_label} must have at least one Tab") % {:dialog_label => label})
     end
 
+    if dialog_fields.collect(&:name).uniq.count < dialog_fields.size && dialog_fields.collect(&:id).uniq.count >= dialog_fields.size
```

Comment:
The `dialog_fields` method is an expensive operation as it has to traverse tabs and groups, therefore it would be good to reduce the number of times it gets called here.

Maybe a better approach is to collect all the dialog names into an array and look for duplicates.

Example from https://stackoverflow.com/questions/8921999/ruby-how-to-find-and-return-a-duplicate-value-in-array:
```ruby
ary = ["A", "B", "C", "B", "A"]
ary.group_by{ |e| e }.select { |k, v| v.size > 1 }.map(&:first)
# => ["A", "B"]
```

Also, I would suggest we move this logic into a private method that just returns the names (or empty array) of the duplicate fields.